azurerm/internal/services/storage/tests/resource_arm_storage_data_lake_gen2_path_test.go

```diff
@@ -0,0 +1,404 @@
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMStorageDataLakeGen2PathDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMStorageDataLakeGen2Path_withACLWithSpecificUserAndDefaults(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMStorageDataLakeGen2PathExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMStorageDataLakeGen2Path_withOwner(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_data_lake_gen2_path", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMStorageDataLakeGen2PathDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMStorageDataLakeGen2Path_withOwner(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMStorageDataLakeGen2PathExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+func testCheckAzureRMStorageDataLakeGen2PathExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Storage.ADLSGen2PathsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		fileSystemName := rs.Primary.Attributes["filesystem_name"]
+		path := rs.Primary.Attributes["path"]
+		storageID, err := parsers.ParseAccountID(rs.Primary.Attributes["storage_account_id"])
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.GetProperties(ctx, storageID.Name, fileSystemName, path, paths.GetPropertiesActionGetStatus)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Path %q in File System %q (Account %q) does not exist", path, fileSystemName, storageID.Name)
+			}
+
+			return fmt.Errorf("Bad: Get on ADLSGen2PathsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMStorageDataLakeGen2PathDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Storage.ADLSGen2PathsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_storage_data_lake_gen2_path" {
+			continue
+		}
+
+		fileSystemName := rs.Primary.Attributes["filesystem_name"]
+		path := rs.Primary.Attributes["path"]
+		storageID, err := parsers.ParseAccountID(rs.Primary.Attributes["storage_account_id"])
+		if err != nil {
+			return err
+		}
+
+		props, err := client.GetProperties(ctx, storageID.Name, fileSystemName, path, paths.GetPropertiesActionGetStatus)
+		if err != nil {
+			return nil
+		}
+
+		return fmt.Errorf("Path still exists: %+v", props)
```

Comment:
This will need unpacking for something useful, or there will just be a memory address and a collection of empty fields.

Reply:
I think the `path` would be more useful than the `props` there - will amend